### DISCUSSION TITLE
Serialise specific properties only

### DISCUSSION THREAD

**Discussion by lonix1** [aaubry/YamlDotNet#718] [Q&A] [answered]
Supposing these types:
````cs
public class Parent
{
  public int Foo { get; set; }
  public bool Bar { get; set; }
}

public class Chid
{
  public Parent Parent { get; set; }
  public string Baz { get; set; }
  public int Qux { get; set; }                // serialise this
  public string Spam { get; set; }
  public bool Ham { get; set; }               // serialise this
  public double Eggs { get; set; }            // serialise this
}
````

If I serialise `Child`, it will include its entire object graph - all public properties, including `Parent`.

How do I serialise specific properties only? For example: `Qux`, `Ham` and `Eggs`, **only**.

(I don't want to specify which properties to "exclude", but rather those to "include".)

**Comment by EdwardCooke** [comment, marked as answer]
To restate what you're looking for, to make sure I understand clearly, is to make it so by default nothing would get serialized unless there was an attribute on the property? The opposite of the way it currently is. Is that right?

I believe the only way we can exclude things is by decorating the properties you want to exclude with the `YamlIgnore` attribute.

**Reply by lonix1** [reply]
Yes that's right. But that's too messy and interferes with my domain, so I went in a different direction - I created a DTO and serialise that instead. Thanks.
